csharp/src/Ice/CollocatedRequestHandler.cs

```diff
@@ -216,21 +224,23 @@ private async ValueTask InvokeAllAsync(OutgoingRequestFrame outgoingRequest, int
             }
         }
 
-        private void SendResponse(int requestId, OutgoingResponseFrame responseFrame)
+        private void SendResponse(int requestId, OutgoingResponseFrame outgoingResponseFrame)
         {
             OutgoingAsyncBase? outAsync;
             lock (this)
             {
                 var responseBuffer = new ArraySegment<byte>(VectoredBufferExtensions.ToArray(
-                        Ice1Definitions.GetResponseData(responseFrame, requestId)));
+                        Ice1Definitions.GetResponseData(outgoingResponseFrame, requestId)));
+                ArraySegment<byte> header = responseBuffer.Slice(0, Ice1Definitions.HeaderSize + 4);
+                responseBuffer = responseBuffer.Slice(Ice1Definitions.HeaderSize + 4);
+                var incomingResponseFrame = new IncomingResponseFrame(_adapter.Communicator, responseBuffer);
```

Comment:
Same here, we could call TraceColloatedFrame. 

It's also not clear why the incoming response frame isn't constructed using the outgoing response frame like we do for the incoming/outgoing request frame. There's no need to create a header in the collocated case.